test/unit/test_put_get.py

```diff
@@ -70,6 +70,38 @@ def test_put_error(tmpdir):
     chmod(file1, 0o700)
 
 
+@pytest.mark.skipif(IS_WINDOWS, reason="permission model is different")
+def test_get_empty_file(tmpdir):
```

Comment:
I'd like an integration test that actually tries to download a non-existent file for all cloud providers instead of a unit test.
That way we would know for sure that we handled all the different cases for all providers.